src/deltalake_ext.rs

```diff
@@ -297,6 +297,231 @@ impl DeltaWriter {
+        }
+        DataType::Int64 => {
+            let min_array = as_primitive_array::<arrow::datatypes::Int64Type>(&min_array);
+            let min = arrow::compute::min(min_array);
+            let min = min.map(|i| Value::Number(Number::from(i)));
+
+            let max_array = as_primitive_array::<arrow::datatypes::Int64Type>(&max_array);
+            let max = arrow::compute::max(max_array);
+            let max = max.map(|i| Value::Number(Number::from(i)));
+
+            Ok((min, max))
+        }
+        DataType::Float32 => {
+            let min_array = as_primitive_array::<arrow::datatypes::Float32Type>(&min_array);
+            let min = arrow::compute::min(min_array);
+            let min = min
+                .map(|f| Number::from_f64(f as f64).map(|n| Value::Number(n)))
+                .flatten();
+
+            let max_array = as_primitive_array::<arrow::datatypes::Float32Type>(&max_array);
+            let max = arrow::compute::max(max_array);
+            let max = max
+                .map(|f| Number::from_f64(f as f64).map(|n| Value::Number(n)))
+                .flatten();
+
+            Ok((min, max))
+        }
+        DataType::Float64 => {
+            let min_array = as_primitive_array::<arrow::datatypes::Float64Type>(&min_array);
+            let min = arrow::compute::min(min_array);
+            let min = min
+                .map(|f| Number::from_f64(f).map(|n| Value::Number(n)))
+                .flatten();
+
+            let max_array = as_primitive_array::<arrow::datatypes::Float64Type>(&max_array);
+            let max = arrow::compute::max(max_array);
+            let max = max
+                .map(|f| Number::from_f64(f).map(|n| Value::Number(n)))
+                .flatten();
+
+            Ok((min, max))
+        }
+        _ => Ok((None, None)),
+    }
+}
+
+fn arrow_array_from_bytes(
+    data_type: DataType,
+    capacity: usize,
+    byte_arrays: Vec<&[u8]>,
+) -> Arc<dyn Array> {
+    let mut buffer = MutableBuffer::new(capacity);
+
+    for arr in byte_arrays.iter() {
+        buffer.extend_from_slice(arr);
+    }
+
+    let builder = ArrayData::builder(data_type)
+        .len(byte_arrays.len())
+        .add_buffer(buffer.into());
+
+    let data = builder.build();
+
+    make_array(data)
+}
+
+fn min_max_strings_from_stats(
+    stats_with_min_max: &Vec<&Statistics>,
+) -> (Option<Value>, Option<Value>) {
+    let min_string_candidates = stats_with_min_max
+        .iter()
+        .filter_map(|s| str_opt_from_bytes(s.min_bytes()));
+
+    let min_value = min_string_candidates
+        .min()
+        .map(|s| Value::String(s.to_string()));
+
+    let max_string_candidates = stats_with_min_max
+        .iter()
+        .filter_map(|s| str_opt_from_bytes(s.max_bytes()));
+
+    let max_value = max_string_candidates
+        .max()
+        .map(|s| Value::String(s.to_string()));
+
+    return (min_value, max_value);
+}
+
+fn is_utf8(opt: Option<LogicalType>) -> bool {
```

Comment:
For tiny helper functions like this, we could add inline annotation to help guide the compiler.

```suggestion
[inline]
fn is_utf8(opt: Option<LogicalType>) -> bool {
```